ang/test/civicase/CaseListDirectives.spec.js

```diff
@@ -151,18 +157,195 @@ describe('CaseListDirective', function () {
+          expect(CRM.$.fn.addClass).not.toHaveBeenCalledWith('civicase__case-list-sortable-header');
+        });
+      });
+      describe('when sortable is true and header is not blank', function () {
+        describe('basic tests', function () {
+          beforeEach(function () {
+            scope.sort = { field: '', dir: '', sortable: true };
+            header = 'next_activity';
+            compileDirective(scope, header);
+          });
+
+          it('makes the header sortable', function () {
+            expect(CRM.$.fn.addClass).toHaveBeenCalledWith('civicase__case-list-sortable-header');
+          });
+        });
+
+        describe('headerClickEventHandler', function () {
+          describe('when the clicked header is already sorted', function () {
+            beforeEach(function () {
+              header = 'next_activity';
+              scope.sort = { field: header, dir: '', sortable: true };
+              scope.changeSortDir = jasmine.createSpy('changeSortDir');
+              compileDirective(scope, header);
+              element.trigger('click');
+            });
+
+            it('changes the sorting direction', function () {
+              expect(scope.changeSortDir).toHaveBeenCalled();
+            });
+          });
+
+          describe('when the clicked header is not already sorted', function () {
+            beforeEach(function () {
+              header = 'next_activity';
+              scope.sort = { field: 'not_next_activity', dir: '', sortable: true };
+              scope.changeSortDir = jasmine.createSpy('changeSortDir');
+              compileDirective(scope, header);
+              element.trigger('click');
+            });
+
+            it('sorts the clicked header', function () {
+              expect(scope.sort.field).toBe(header);
+            });
+
+            it('sorts the clicked header in ascending order', function () {
+              expect(scope.sort.dir).toBe('ASC');
+            });
+          });
+        });
+
+        describe('sortWatchHandler', function () {
+          beforeEach(function () {
+            header = 'next_activity';
+            scope.sort = { field: header, dir: '', sortable: true };
+            scope.changeSortDir = jasmine.createSpy('changeSortDir');
+            compileDirective(scope, header);
+          });
+
+          describe('when the clicked header is already sorted', function () {
+            beforeEach(function () {
+              scope.sort = { field: header, dir: 'ASC', sortable: true };
+              scope.$digest();
+            });
+
+            it('changes the sorting icon direction', function () {
+              expect(element.hasClass('active')).toBe(true);
+            });
+
+            it('removes the sorting icon before adding a new one', function () {
+              expect(CRM.$.fn.find).toHaveBeenCalledWith('.civicase__case-list__header-toggle-sort');
+              expect(CRM.$.fn.remove).toHaveBeenCalled();
```

Comment:
this is where testing on the actual element is better. The code could do this:

```js
element.find('.civicase__case-list__header-toggle-sort');
element.find('.something-else').remove();
```

and the test would pass even though `.something-else` was what actually was removed.

There is a simpler way to find if the element was removed:

```js
expect(element.find('.civicase__case-list__header-toggle-sort').length).toBe(0);
```